test/e2e/main_test.go

```diff
@@ -0,0 +1,90 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	kconf "sigs.k8s.io/e2e-framework/klient/conf"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"testing"
+)
+
+var (
+	testEnv       env.Environment
+	cloudProvider string
+	provisioner   CloudProvision
+)
+
+func TestMain(m *testing.M) {
+	var err error
+
+	// CLOUD_PROVIDER is required.
+	cloudProvider = os.Getenv("CLOUD_PROVIDER")
+	if cloudProvider == "" {
+		fmt.Println("CLOUD_PROVIDER should be exported in the environment")
+		os.Exit(1)
+	}
+
+	// Create an empty test environment. At this point the client cannot connect to the cluster
+	// unless it is running with an in-cluster configuration.
+	testEnv = env.New()
+
+	// In case TEST_E2E_PROVISION is exported then it will try to provision the test environment
+	// in the cloud provider. Otherwise, assume the developer did setup the environment and it will
+	// look for a suitable kubeconfig file.
+	shouldProvisionCluster := false
+	if os.Getenv("TEST_E2E_PROVISION") != "" {
```

Comment:
Not necessarily a complaint, just an observation. I originally tried getting the cluster provisioned and it failed (probably due to user error), so I created it myself and then ran `TEST_E2E_PROVISION=no CLOUD_PROVIDER=libvirt make test-e2e` thinking that `no` would disable provisioning, yet hit a bunch of errors still. Having checked this code I can see that it's not checking for `TEST_E2E_PROVISION=yes`, but for it set at all, so if we keep this behaviour it is probably worth documenting?

Reply:
I will changed it to consider `TEST_E2E_PROVISION=no` case too. thanks!

Reply:
```
diff --git a/test/e2e/main_test.go b/test/e2e/main_test.go
index fc04fd1..5d33253 100644
--- a/test/e2e/main_test.go
+++ b/test/e2e/main_test.go
@@ -34,7 +34,7 @@ func TestMain(m *testing.M) {
        // in the cloud provider. Otherwise, assume the developer did setup the environment and it will
        // look for a suitable kubeconfig file.
        shouldProvisionCluster := false
-       if os.Getenv("TEST_E2E_PROVISION") != "" {
+       if os.Getenv("TEST_E2E_PROVISION") == "yes" {
                shouldProvisionCluster = true
        }

```

Reply:
Fixed.